lib/src/components/NxCollapsibleItemsSelect/NxCollapsibleMultiSelect/__tests__/NxCollapsibleMultiSelect.test.tsx

```diff
@@ -4,355 +4,559 @@
+          const trigger = renderAndGetTrigger({
+            options: [],
+            tooltipModifierClass: 'tooltipClass'
+          });
+
+          await runTimers();
+          expect(screen.queryByRole('tooltip')).not.toBeInTheDocument();
+
+          fireEvent.mouseOver(trigger);
+          await runTimers();
+
+          const tooltip = screen.getByRole('tooltip');
+
+          expect(tooltip).toHaveTextContent('There are no foobar options');
+          expect(tooltip.querySelector('.tooltipClass')).toBeInTheDocument();
+        });
       });
-      const renderOption = shallowRender.renderProp('renderOption');
 
-      expect(renderOption({id: 'foo', name: 'Foo'})).toHaveProp('disabled', true);
-      expect(renderOption({id: 'bar', name: 'Bar'})).toHaveProp('disabled', true);
-      expect(renderOption({id: null, name: 'Null'})).toHaveProp('disabled', true);
+      describe('when component is disabled explicitly', function() {
+        it('has a customized text as specified by disabledTooltip prop', async function() {
+          const trigger = renderAndGetTrigger({
+            disabled: true,
+            disabledTooltip: 'No options'
+          });
+
+          await runTimers();
+          expect(screen.queryByRole('tooltip')).not.toBeInTheDocument();
+
+          fireEvent.mouseOver(trigger);
+          await runTimers();
+
+          const tooltip = screen.getByRole('tooltip');
+
+          expect(tooltip).toHaveTextContent('No options');
+        });
+
+        it('has specified classname when tooltipModifierClass prop is provided', async function() {
+          const trigger = renderAndGetTrigger({
+            disabled: true,
+            disabledTooltip: 'No options',
+            tooltipModifierClass: 'tooltipClass'
+          });
+
+          await runTimers();
+          expect(screen.queryByRole('tooltip')).not.toBeInTheDocument();
+
+          fireEvent.mouseOver(trigger);
+          await runTimers();
+
+          const tooltip = screen.getByRole('tooltip');
+
+          expect(tooltip).toHaveTextContent('No options');
+          expect(tooltip.querySelector('.tooltipClass')).toBeInTheDocument();
+        });
+
+        it('does not show when disabledTooltip prop is not provided', async function() {
+          const trigger = renderAndGetTrigger({ disabled: true });
+
+          await runTimers();
+          expect(screen.queryByRole('tooltip')).not.toBeInTheDocument();
+
+          fireEvent.mouseOver(trigger);
+          await runTimers();
+
+          expect(screen.queryByRole('tooltip')).not.toBeInTheDocument();
+        });
+      });
     });
+  });
 
-    it('uses option name as checkbox label', function () {
-      const renderOption = getShallow().renderProp('renderOption');
+  describe('filter', function() {
+    const filterView = (props?: Partial<Props>) => quickRender({
+      ...props,
+      filterThreshold: 2,
+      onFilterChange: jest.fn()
```

Comment:
Set up mocks in individual tests or `beforeEach`
```suggestion
      onFilterChange: () => {}
```

Reply:
Updated here 90a6389ea72a7c8eb19b4ef11c30b29abeb126be